src/connector/src/sink/iceberg/compaction.rs

```diff
@@ -46,10 +57,124 @@ impl NimtableCompactionConfig {
+                        )
+                    })?,
+            ),
         })
     }
 }
 
+async fn run_compact(
+    database: String,
+    table: String,
+    catalog_config: HashMap<String, String>,
+    mut commit_rx: mpsc::UnboundedReceiver<()>,
+) {
+    let config = match NimtableCompactionConfig::from_env() {
+        Ok(config) => config,
+        Err(e) => {
+            error!(database, table, e = ?e.as_report(), "failed to start compact worker");
+            return;
+        }
+    };
+    let compact_frequency = config.compact_frequency;
+    let min_compact_gap_duration_sec = config.min_compact_gap_duration_sec;
+    let mut pending_commit_num = 0;
+    let client = NimtableCompactionClient::new(config).await;
+    let new_backoff = || {
+        ExponentialBackoff::from_millis(1000)
+            .factor(2)
+            .max_delay(Duration::from_secs(60))
+    };
+
+    let mut prev_compact_success_time = Instant::now();
+    let mut error_backoff = new_backoff();
+    let mut error_count = 0;
+
+    loop {
+        if pending_commit_num >= compact_frequency
+            && Instant::now().duration_since(prev_compact_success_time)
+                > min_compact_gap_duration_sec
+        {
+            let compact_start_time = Instant::now();
+            if let Err(e) = client
+                .compact(database.clone(), table.clone(), &catalog_config)
+                .await
+            {
+                let backoff_duration = error_backoff.next().expect("should exist");
+                error_count += 1;
+                error!(
+                    err = ?e.as_report(),
+                    ?backoff_duration,
+                    error_count,
+                    database, table, "failed to compact"
+                );
+                sleep(backoff_duration).await;
+            } else {
+                info!(database, table, elapsed = ?compact_start_time.elapsed(),  "compact success");
+                pending_commit_num = 0;
+                error_backoff = new_backoff();
+                error_count = 0;
+                prev_compact_success_time = Instant::now();
+            }
+        }
+        if commit_rx.recv().await.is_some() {
+            pending_commit_num += 1;
+        } else {
+            break;
+        }
+    }
+}
+
+pub fn spawn_compaction_client(
+    config: &IcebergConfig,
+) -> anyhow::Result<(mpsc::UnboundedSender<()>, oneshot::Sender<()>)> {
+    let warehouse_path = config.common.warehouse_path.clone();
+    let database = config
+        .common
+        .database_name
+        .clone()
+        .ok_or_else(|| anyhow!("should specify database"))?;
+    let table = config.common.table_name.clone();
+    let (commit_tx, commit_rx) = mpsc::unbounded_channel();
+    let (finish_tx, finish_rx) = oneshot::channel();
+
+    let _join_handle = tokio::spawn(async move {
+        let catalog_config = HashMap::from_iter([
+            ("type".to_string(), "hadoop".to_string()),
```

Comment:
Let's change it to the JDBC catalog. https://github.com/risingwavelabs/risingwave/blob/5bac8e7d0485ab0b0efabc03c148220bc294bc2b/integration_tests/iceberg-sink2/docker/jdbc/spark-script/spark-connect-server.sh#L2-L1

Reply:
Changed to extract the catalog config from IcebergConfig. Supported catalog type is `storage` and `jdbc`.